endpoints/internal-check-file-acl.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Automattic\VIP\Files\Acl;
+
+require_once __DIR__ . '/../files/acl/pre-wp-utils.php';
+
+$file_request_uri = $_SERVER['HTTP_X_ORIGINAL_URI'] ?? null;
+
+if ( ! $file_request_uri ) {
+	trigger_error( 'VIP Files ACL failed due to empty URI', E_USER_WARNING );
+
+	http_response_code( 500 );
+
+	exit;
+}
+
+$file_path = parse_url( $file_request_uri, PHP_URL_PATH );
+
+$is_valid_path = Pre_WP_Utils\validate_path( $file_path );
+if ( ! $is_valid_path ) {
+	http_response_code( 500 );
+
+	exit;
+}
+
+list( $subdirectory, $sanitized_file_path ) = Pre_WP_Utils\sanitize_and_split_path( $file_path );
+
+if ( $subdirectory ) {
+	$_SERVER['REQUEST_URI'] = $subdirectory . ( $_SERVER['REQUEST_URI'] ?? '' );
```

Comment:
I'm not 100% sure this is safe. What if `$subdirectory` was a full url, like `//foo.com/path` (which will pass the "starts with `/` check"), or included path traversal characters (`../`)? A hypothetical request would be: `http://bar.com/%2F%2Ffoo.com/path/wp-content/uploads/baz` (though the web server may correctly handle that). Code elsewhere may not correctly (or safely) handle weird `REQUEST_URI` values. We should do more sanitization inside `sanitize_and_split_path()`.

Reply:
Suggestions on how to sanitize this? We are limited to pure PHP and it should ideally be fast.

I worry that this might become whack-a-mole like since it's pretty open in terms what is allowed in a URL.